sql/plan/exchange.go

```diff
@@ -263,17 +271,29 @@ func (it *exchangeRowIter) Next() (sql.Row, error) {
 	}
 }
 
-func (it *exchangeRowIter) Close() (err error) {
-	if it.quit != nil {
-		close(it.quit)
-		it.quit = nil
+func (it *exchangeRowIter) quit() chan struct{} {
+	it.quitMut.RLock()
+	defer it.quitMut.RUnlock()
+	if it.quitChan == nil {
+		return nil
+	}
+
+	return it.quitChan
```

Comment:
Isn't easier here to just
```go
return it.quitChan
```
instead of
```go
if it.quitChan == nil {
    return nil
}
return it.quitChan
```
?

Reply:
True, fixed